src/scripts/DataLayerItem.js

```diff
@@ -28,6 +27,94 @@ const itemType = {
   LISTENER_OFF: 'listenerOff'
 };
 
+/**
+ * Constraints for each type of the item configuration.
+ */
+const constraints = {
+  data: {
+    properties: [
+      {
+        key: 'data',
+        type: 'object',
+        mandatory: true
+      }
+    ],
+    customPropertiesAllowed: false
```

Comment:
if custom properties are not allowed across the board, I would remove this config for now.